src/main/scala/org/opennetworkinsight/dns/DNSPostLDA.scala

```diff
@@ -142,8 +143,28 @@ object DNSPostLDA {
     addcol("score")
 
     logger.info("Persisting data")
-    val scored = src_scored.filter(elem => elem._1 < threshold).sortByKey().map(row => row._2.mkString(","))
-    scored.saveAsTextFile(resultsFilePath)
+
+
+    val filtered = src_scored.filter(elem => elem._1 < threshold)
+
+    val count = filtered.count
+
+    val takeCount  = if (topK == -1 || count < topK) {
+      count.toInt
+    } else {
+      topK
+    }
+    class OrderByLeastScoreToTop() extends Ordering[(Double,Array[Any])] {
+      def compare(p1: (Double, Array[Any]), p2: (Double, Array[Any]))    = p2._1.compare(p1._1)
+    }
+
+    implicit val ordering = new OrderByLeastScoreToTop()
+
+    val top : Array[(Double,Array[Any])] = filtered.top(takeCount)
```

Comment:
We need to use _takeOrdered_ rather than _top_. Top is going to take the largest elements from the RDD while takeOrdered is going to take the K elements based on ascending order. 
![screen shot 2016-08-16 at 10 56 50 am](https://cloud.githubusercontent.com/assets/17070509/17706209/c1238bf8-63a0-11e6-87ac-d6f38aa73e38.png)